requirements.txt

```diff
@@ -7,6 +7,7 @@ ply==3.4
 slimit==0.8.1
 six==1.9.0
 django-appconf==1.0.1
+peewee==2.6.3
```

Comment:
Sounds like it has good benefits, and it looks pure-Python with some kind
of [Cython speed-ups](
https://github.com/coleifer/peewee/blob/master/setup.py#L10) that I don't
understand how works.. should probably test that it builds a cross-platform
result.

But if there aren't any critical optimization benefits, perhaps its better
to wait until Kolibri with introducing a whole new ORM?

I don't think KA Lite or Kolibri should ever have 2 ORMs.... would be a
nightmare to be honest... should we then define models for both ORMs? How
do we create a relation from one ORM to the other? How does query caching
work across ORMs? Who's responsible for migrations? etc.

On 15 September 2015 at 01:09, Richard Tibbles <notifications@github.com>
wrote:

> In requirements.txt
> <https://github.com/learningequality/ka-lite/pull/4261#discussion_r39458535>
> :
>
> > @@ -7,6 +7,7 @@ ply==3.4
> >  slimit==0.8.1
> >  six==1.9.0
> >  django-appconf==1.0.1
> > +peewee==2.6.3
>
> This handles the database backend for the content caches. It is a light
> wrapper around SQLite, inspired by the Django ORM.
>
> So far it seems pretty efficient and useable, and I've not had any of the
> issues that @MCGallaspy <https://github.com/MCGallaspy> had with using
> the Django ORM for other content ORMing.
>
> It also allows very easy hot swapping of database files during runtime,
> for, e.g., using different databases for different languages (and also
> channels).
>
> That said, I'm not wedded to it, so if we can do this efficiently without
> adding a dependency, happy with that too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/learningequality/ka-lite/pull/4261/files#r39458535>.
>